### PR TITLE
DOC: add API docs + pre-release note scripts

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Add auto-generated API documentation and pre-release note scripts
* Unpin line-profiler for Python 3.11 support

## Motivation and Context
There are no docs here!

## How Has This Been Tested?
Locally built docs

## Where Has This Been Documented?
This is docs

## Screenshots (if appropriate):
image
image

### CODE REVIEW

File: docs/source/upcoming_release_notes/51-add-get-info-json.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Features
+--------
+- Add cleaned up and performant version of heavily-used ``get_info`` in
+  :mod:`pcdsutils.info`.
+
+Device Updates
+--------------
+- N/A
+
```
Comment (tangkong) [inline docs/source/upcoming_release_notes/51-add-get-info-json.rst:16]:
```suggestion
```

File: docs/source/upcoming_release_notes/template-full.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  provide a new base classes, add options to helper methods, etc.
+
+Device Updates
+--------------
+- List new features or specific changes for individual devices.
+
+New Devices
+-----------
+- List new device support that is being added.
+
```
Comment (tangkong) [inline docs/source/upcoming_release_notes/template-full.rst:28]:
```suggestion
```

File: docs/source/upcoming_release_notes/template-short.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- N/A
+
+Features
+--------
+- N/A
+
+Device Updates
+--------------
+- N/A
+
```
Comment (tangkong) [inline docs/source/upcoming_release_notes/template-short.rst:15]:
```suggestion
```

pcdsutils doesn't have devices right?  (unless I'm forgetting something)

**Review by tangkong — APPROVED** [review]
LGTM

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**